docs/source/worker-environments.rst

```diff
@@ -0,0 +1,84 @@
+Worker Environments
+===================
+
+A "worker environment" can be used to identify and prepare a set of workers for
+certain tasks. For example, you may have some setup logic to connect to a
+database that must be run prior to doing work, but don't want to open a
+connection on every single worker. As another example, you may have certain
+tasks that require a high amount of memory, so you wish to identify workers with
+at least 32GB of RAM.
+
+Components
+----------
+
+An environment consists of
+
+1. A unique name like ``"high-memory"`` or ``"database-access"``
+2. A ``condition``, a function of no arguments to be called on the worker.
+   Determines whether that worker qualifies for the environment.
+3. A ``setup`` a function of no arguments to be run on the worker when they
+   join an environment
+4. A ``teardown`` function of no arguments to be run on the worker when they
+   leave the environment
+5. A ``state`` variable (TODO: what exactly is this? Tentatively, each worker
+   has a dictionary ``environment_state :: name -> Dict``, where you can get and
+   put whatever you want in that nested ``Dict``)
```

Comment:
Is having state necessary?  Perhaps drop this but include examples of environment instances with state?